### ISSUE TITLE
sftp port spec is broken for some versions

### ISSUE THREAD

**Issue by andre-merzky** [radical-cybertools/saga-python#604]
```mail
On Thu, Jan 12, 2017 at 8:56 PM, Mark Perri <mark.perri@sonoma.edu> wrote:
> I'm trying to use a non-standard port to submit jobs through slurm+ssh, but
> there is a problem when SAGA uploads the SLURM script. (version 0.44)
>
> It looks like SAGA is using /usr/bin/sftp -P 2222 to set the port, but my
> sftp specifies to use sftp -oPort=2222 to set the port number.

Interesting - we did not yet encounter an sftp version which does not
accept '-P [port]'.  May I ask what sftp version you are using, and on
what system you encountered it?  On any debian-derived Linux system
(like Ubuntu), you can do that with:

  $ dpkg -S `which sftp`
  openssh-client: /usr/bin/sftp

  $ dpkg -l | grep openssh-client
  ii  gsi-openssh-clients
      7.1p2f-4+gt6.xenial                         
      amd64        
      GSI OpenSSH Clients
  ii  openssh-client
      1:7.3p1-1
      amd64
      secure shell (SSH) client, for secure access to remote machines

  $ cat /etc/lsb-release
  DISTRIB_ID=Ubuntu
  DISTRIB_RELEASE=16.10
  DISTRIB_CODENAME=yakkety
  DISTRIB_DESCRIPTION="Ubuntu 16.10"


> Line 646 in utils/pty_shell_factory.py should read:
> info['sftp_args'] += "-oPort=%d " % int(url.port)

We need to check if that works for the different sftp versions around,
but indeed, that would be the alternative.

Thanks, Andre.
```

**Comment by andre-merzky** [comment]
This is addressed in #605, but waits for confirmation.

**Comment by andre-merzky** [comment]
This is considered fixed.
